packages/core/src/components/BaseList/BaseList.tsx

```diff
@@ -0,0 +1,164 @@
+import React, { forwardRef, type ReactElement, useMemo, useRef, useState } from "react";
+import cx from "classnames";
+import useMergeRef from "../../hooks/useMergeRef";
+import useIsomorphicLayoutEffect from "../../hooks/ssr/useIsomorphicLayoutEffect";
+import { type BaseListProps } from "./BaseList.types";
+import { BaseListContext, type BaseListContextProps } from "./context/BaseListContext";
+import { useBaseListFocus } from "./hooks/useBaseListFocus";
+import { useBaseListKeyboard } from "./hooks/useBaseListKeyboard";
+import {
+  generateListId,
+  getChildRole,
+  getItemComponentType,
+  isListItem,
+  mergeStyleWithMaxHeight
+} from "./utils/BaseListUtils";
+import styles from "./BaseList.module.scss";
+
+const BaseList = forwardRef(
+  (
+    {
+      className,
+      id,
+      component = "ul",
+      children,
+      ariaLabel,
+      ariaDescribedBy,
+      "aria-controls": ariaControls,
+      role = "listbox",
+      size = "medium",
+      maxHeight,
+      tabIndex = 0,
```

Comment:
do we really need it? I think it should be "-1" by default and not to allow change it